cmd/gh-classroom/shared/shared.go

```diff
@@ -109,30 +111,51 @@ func ListAcceptedAssignments(client api.RESTClient, assignmentID int, page int,
 	return acceptedAssignmentList, nil
 }
 
+type assignmentList struct {
+	assignments []classroom.AcceptedAssignment
+	Error       error
+}
+
 func ListAllAcceptedAssignments(client api.RESTClient, assignmentID int, perPage int) (classroom.AcceptedAssignmentList, error) {
-	var page = 1
-	response, err := classroom.GetAssignmentList(client, assignmentID, page, perPage)
+
+	//Calculate the number of go channels to create. We will assign 1 channel per page
+	//so we don't put too much pressure on the API all at once
+	assignment, err := classroom.GetAssignment(client, assignmentID)
 	if err != nil {
 		return classroom.AcceptedAssignmentList{}, err
 	}
-
-	if len(response) == 0 {
-		return classroom.AcceptedAssignmentList{}, nil
-	}
-
-	//keep calling getAssignmentList until we get them all
-	var nextList []classroom.AcceptedAssignment
-	for hasNext := true; hasNext; {
-		page += 1
-		nextList, err = classroom.GetAssignmentList(client, assignmentID, page, perPage)
-		if err != nil {
-			return classroom.AcceptedAssignmentList{}, err
+	tmp := math.Ceil(float64(assignment.Accepted) / float64(perPage))
+	numChannels := int(tmp)
+
+	ch := make(chan assignmentList)
+	var wg sync.WaitGroup
+	wg.Add(numChannels)
+	for page := 1; page <= numChannels; page++ {
+		go func(pg int) {
+			defer wg.Done()
+			response, err := classroom.GetAssignmentList(client, assignmentID, pg, perPage)
+			ch <- assignmentList{
+				assignments: response,
+				Error:       err,
+			}
+		}(page)
+	}
+
```

Comment:
We will want to wait to read from the channel until all goroutines are complete.
```suggestion
wg.Wait()
```